src/index.js

```diff
@@ -217,28 +217,35 @@ function UseWalletProvider({
     [chainId, connectorsInitsOrConfigs]
   )
 
-  const deactivate = useCallback(async () => {
+  const reset = useCallback(() => {
     if (web3ReactContext.active) {
-      await web3ReactContext.deactivate()
+      web3ReactContext.deactivate()
     }
-    setActivating(null)
-    setActivated(null)
+    setConnector(null)
+    setError(null)
+    setStatus('disconnected')
   }, [web3ReactContext])
 
-  const activate = useCallback(
+  const connect = useCallback(
     async (connectorId = 'injected') => {
       // Prevent race conditions between connections by using an external ID.
       const id = ++activationId.current
 
-      deactivate()
+      reset()
 
       // Check if another connection has happened right after deactivate().
       if (id !== activationId.current) {
         return
       }
 
+      // If no connection happene, we're in the right context, and we can safely update
+      // the connection stage status
+      setStatus('connecting')
+
       if (!connectors[connectorId]) {
-        throw new UnsupportedConnectorError(connectorId)
+        setStatus('error')
+        setError(UnsupportedConnectorError(connectorId))
```

Comment:
Should we update the naming of these errors per [this suggestion](https://github.com/aragon/use-wallet/issues/28#issuecomment-600036707)?

Reply:
I think it's a good renaming, that way we don't have a prefix + suffix, just the later. What do you think @bpierre ?

Reply:
Oh sorry I missed this comment, yes I think these are better names! :+1: